### ISSUE TITLE
Custom results handling logic

### ISSUE THREAD

**Issue by lukaville** [MarathonLabs/marathon#194]
We want to check the flakiness rate using counting sharding strategy.
Currently, marathon marks test as successful if any of the test runs were successful.
We need to have the ability to customise this behaviour. For example, we should be able to fail the test if any of the runs were failed.

Probably, we can introduce an interface that returns the final test result based on the test history. Something like this:
```
interface ResultStrategy {
    fun process(results: List<Pair<DeviceInfo, TestResult>>): Pair<DeviceInfo, TestResult>
}
```

**Comment by tagantroy** [comment]
It can be implemented as a plugin or #196, what do you think?

**Comment by lukaville** [comment]
Probably strict mode will be enough. Looks like other strategies not so useful
